src/openshift/cluster.ts

```diff
@@ -275,9 +276,34 @@ export class Cluster extends OpenShiftItem {
 
         if (response !== 'Yes') return null;
 
-        const clusterURL = await Cluster.getUrl();
+        let clusterIsUp = false;
+        let clusterURL: string;
 
-        if (!clusterURL) return null;
+        do {
+            clusterURL = await Cluster.getUrl();
+
+            if (!clusterURL) return null;
+
+            try {
+                // disable cert checking. crc uses a self-signed cert,
+                // which means this request will always fail on crc unless cert checking is disabled
+                const clusterApiResponse = await fetch(clusterURL, {
+                    strictSSL: false
+                });
+                // don't check for 2XX status, since we are expecting an "unauthorized" response from the server
+                if (500 <= clusterApiResponse.status && clusterApiResponse.status < 600) {
+                    void window.showWarningMessage(
+                        'Received 5XX response (server error) from the cluster API server',
+                    );
```

Comment:
Is there documentation somewhere that states a cluster will always return a non-500 error if it is up ? I would think as long as `clusterApiReponse` has any value, it would indicate the cluster is up.

Reply:
Yeah, I haven't been able to replicate a case where the API returns 5XX when it's starting up, so I can probably remove this check